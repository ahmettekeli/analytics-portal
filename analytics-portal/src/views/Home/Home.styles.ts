import styled from "styled-components";
//TODO Home page styling.
