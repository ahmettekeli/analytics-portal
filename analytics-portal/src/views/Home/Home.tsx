import { motion } from "framer-motion";
import { useNavigate } from "react-router-dom";
import Button from "@material-ui/core/Button";

function Home() {
  const navigate = useNavigate();
  return (
    <motion.div
      initial={{ opacity: 0 }}
      animate={{ opacity: 1 }}
      exit={{ opacity: 0 }}
    >
      <h3>Home page</h3>
      <Button
        variant="contained"
        color="primary"
        onClick={() => {
          navigate("/overview");
        }}
      >
        Overview
      </Button>
    </motion.div>
  );
}

export default Home;
