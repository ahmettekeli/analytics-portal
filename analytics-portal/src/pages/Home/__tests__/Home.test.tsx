import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Home from "../Home";

describe("Home", () => {
  let renderResult: ReturnType<typeof render>;
  beforeEach(() => {
    renderResult = render(
      <Router>
        <Home />
      </Router>
    );
  });
  test("Renders Home component", () => {
    expect(renderResult.getByTestId("Home")).toBeInTheDocument();
  });

  test("Overview button redirects to Overview page", () => {
    const overviewButton = renderResult.getByText("Overview");
    overviewButton.click();
    expect(window.location.pathname).toBe("/overview");
  });

  test("Home matches snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <Home />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
