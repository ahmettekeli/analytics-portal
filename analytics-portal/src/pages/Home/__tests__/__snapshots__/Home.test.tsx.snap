// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home matches snapshot 1`] = `
<div
  data-testid="Home"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <h3>
    Home page
  </h3>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Overview
    </span>
  </button>
</div>
`;
